public-api/src/integration/java/org/pmiops/workbench/google/BuildIapRequest.java

```diff
@@ -0,0 +1,62 @@
+package org.pmiops.workbench.google;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.IdTokenProvider;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.*;
+
+public class BuildIapRequest {
+    private static final String IAM_SCOPE = "https://www.googleapis.com/auth/iam";
+
+    private static final HttpTransport httpTransport = new NetHttpTransport();
+
+    private static String TOKEN = "";
```

Comment:
I'm not sure if this was from experimentation, but the static usage here is a bit sketchy. I think you really just want a single method on the class that fetches a token.

Reply:
Removed thanks.